pandas/core/arrays/arrow/array.py

```diff
@@ -403,7 +419,12 @@ def _cast_pointwise_result(self, values) -> ArrayLike:
             return self[:0].copy()
 
         try:
-            arr = pa.array(values, from_pandas=True)
+            if self.dtype.kind in "iufc" and not is_nan_na():
+                values = np.asarray(values, dtype=object)
```

Comment:
If `values` is a typed array of sorts (`np.ndarray`, `pa.Array`), I suppose we can avoid the cast to object.